core/storage/classifier/gae_models.py

```diff
@@ -83,3 +85,92 @@ def create(
 
         classifier_data_model_instance.put()
         return instance_id
+
+
+class ClassifierTrainingJobModel(base_models.BaseModel):
+    """Model for storing classifier training jobs.
+
+    The id of instances of this class has the form
+    {{exp_id}}.{{random_hash_of_16_chars}}
+    """
+
+    # The ID of the algorithm used to create the model.
+    algorithm_id = ndb.StringProperty(required=True, choices=ALGORITHM_CHOICES,
+                                      indexed=True)
+    # The exploration_id of the exploration to whose state the model belongs.
+    exp_id = ndb.StringProperty(required=True, indexed=True)
+    # The exploration version at the time this training job was created.
+    exp_version_when_created = ndb.IntegerProperty(required=True)
```

Comment:
I think this and state_name should be indexed. I can imagine queries that filter on the triple (exp_id, exp_version, state_name).